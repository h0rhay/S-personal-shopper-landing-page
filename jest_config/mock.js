export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
