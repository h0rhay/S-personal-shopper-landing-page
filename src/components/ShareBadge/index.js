import PropTypes from 'prop-types';
import React from 'react';
import styled, { keyframes } from 'styled-components';
import ShareRoundGraphic from '../../images/share-roundel.svg';

const rotating = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const ShareRoundel = styled.button`
  position: absolute;
  max-width: 8.75rem;
  max-height: 8.75rem;
  display: block;
  border: none;
  background: transparent;
  bottom: -8rem;
  left: 4rem;
  cursor: pointer;
  text-decoration: none;
  > div {
    position: relative;
    > div {
      font-family: 'DINNextLTPro-Light';
      letter-spacing: 1px;
      position: absolute;
      top: calc(50% - 0.75rem);
      left: calc(50% - 1.5rem);
      color: ${({ theme }) => theme.text};
      &:after {
        display: block;
        content: '';
        border-bottom: solid 1px ${({ theme }) => theme.text};
        transform: scaleX(0);
        transition: transform 250ms ease-in-out;
        transform-origin: 0% 50%;
      }
    }
  }
  &:hover {
    > div > div:after {
      transform: scaleX(1);
    }
  }
  @media (max-width: 860px) {
    bottom: -10rem;
    left: unset;
    right: 2rem;
    max-width: 7rem;
    max-height: 7rem;
  }
  path {
    fill: ${({ theme }) => theme.text};
  }
  svg {
    width: 6.5rem;
    animation: 40s ${rotating} linear infinite;
    transform-origin: center;
  }
`;

const ShareBadge = ({ clickHandler }) => (
  <ShareRoundel type="button" role="button" onClick={clickHandler}>
    <div>
      <ShareRoundGraphic />
      <div>SHARE</div>
    </div>
  </ShareRoundel>
);

ShareBadge.propTypes = {
  clickHandler: PropTypes.func,
};

export default ShareBadge;
