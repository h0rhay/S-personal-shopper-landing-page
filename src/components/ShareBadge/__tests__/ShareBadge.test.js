import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import ShareBadge from '../index';

const props = {
  type: 'button',
  onClick: () => {},
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <ShareBadge {...props} />
    </ThemeProvider>
  );

describe('ShareBadge component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme, props });
    expect(container).toMatchSnapshot();
  });
});
