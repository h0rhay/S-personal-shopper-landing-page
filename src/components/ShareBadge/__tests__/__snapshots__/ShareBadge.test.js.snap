// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareBadge component should match screenshot 1`] = `
<div>
  <button
    class="sc-bdvvtL kOTLLx"
    role="button"
    type="button"
  >
    <div>
      <test-file-stub />
      <div>
        SHARE
      </div>
    </div>
  </button>
</div>
`;
